node/resolvers/Mutations/Users.ts

```diff
@@ -612,3 +642,45 @@ const Users = {
 }
 
 export default Users
+
+async function impersonateUser(
+  storefrontPermissionsClient: any,
+  id: string,
+  logger: any,
+  accountName: string,
+  storefrontPermissions: {
+    costcenter: { value: string }
+    userId: { value: string }
+    organization: { value: string }
+    storeUserEmail: { value: string }
+  }
+) {
+  const result = await storefrontPermissionsClient
+    .impersonateUser({ id })
+    .catch((error: any) => {
+      logger.error({
+        error,
+        message: 'impersonateUser-impersonateUserError',
+      })
+
+      return { status: 'error', message: error }
+    })
+
+  const { costcenter, userId, organization, storeUserEmail } =
+    storefrontPermissions
+
+  const metricParams: ImpersonateMetricParams = {
+    account: accountName,
+    target: {
+      costCenterId: costcenter.value,
+      organizationId: organization.value,
+      email: storeUserEmail.value,
+      id: userId.value,
+    },
+    user: undefined, // no information about original user at this point
+  }
+
+  await sendImpersonateUserMetric(metricParams)
```

Comment:
This shouldn't be awaited, right? It should run on the background.

Reply:
Right. Already fixed when looking another comment